Wrappers/WebElement.cs

```diff
@@ -0,0 +1,130 @@
+﻿using OpenQA.Selenium;
+using SeleniumExtras.WaitHelpers;
+using System.Collections.ObjectModel;
+using System.Drawing;
+using Utilities;
+
+namespace Wrappers
+{
+    public class WebPageElement : IWebElement
+
+    {
+        protected IWebDriver Driver;
+        protected IWebElement Element;
+
+        public WebPageElement(IWebElement element)
+        {
+            Element = element;
+        }
+
+        public WebPageElement(IWebDriver driver, By locator)
+        {
+            Driver = driver;
+            Element = Driver.FindElements(locator).FirstOrDefault();
+        }
+
+        public WebPageElement()
+        {
+        }
+
+        public string Text => Element.Text;
+
+        public bool Enabled => Element.Enabled;
+
+        public bool Selected => Element.Selected;
+
+        public bool Displayed => Element.Displayed;
+
+        public string TagName => Element.TagName;
+
+        public Point Location => Element.Location;
+
+        public Size Size => Element.Size;
+
+        public void Clear() => Element.Clear();
+
+        public void Click() => Element.Click();
+
+        public void SendKeys(string text) => Element.SendKeys(text);
+
+        public void Submit() => Element.Submit();
+
+        public string GetAttribute(string attributeName) => Element.GetAttribute(attributeName);
+
+        public string GetDomAttribute(string attributeName) => Element.GetDomAttribute(attributeName);
+
+        public string GetDomProperty(string propertyName) => Element.GetDomProperty(propertyName);
+
+        public string GetCssValue(string propertyName) => Element.GetCssValue(propertyName);
+
+        public ISearchContext GetShadowRoot() => Element.GetShadowRoot();
+
+        public IWebElement FindElement(By by) => Element.FindElement(by);
```

Comment:
Your `FindElement` method returns an instance of usual seleium's WebElement. Instead, you should return your own implementation of `IWebElement` (you should return another instance of your own class `WebPageElement`). You can do this as following:
```
public IWebElement FindElement(By by)
{
    var seleiumWebElement = Element.FindElement(by);
    var myWebElement = new WebPageElement(seleiumWebElement);
    
    return myWebElement;
}
```

Reply:
Ok, returns WebPageElement